lib/spack/spack/ci/__init__.py

```diff
@@ -613,32 +614,47 @@ def copy_stage_logs_to_artifacts(job_spec: spack.spec.Spec, job_log_dir: str) ->
     job_spec, and attempts to copy the files into the directory given
     by job_log_dir.
 
-    Args:
+    Parameters:
         job_spec: spec associated with spack install log
         job_log_dir: path into which build log should be copied
     """
     tty.debug(f"job spec: {job_spec}")
 
     try:
         package_metadata_root = pathlib.Path(spack.store.STORE.layout.metadata_path(job_spec))
+        if not package_metadata_root.is_dir():
+            pkg_cls = spack.repo.PATH.get_pkg_class(job_spec.name)
+            job_pkg = pkg_cls(job_spec)
+
+            package_metadata_root = job_pkg.stage.path
+            archive_files = spack.builder.create(job_pkg).archive_files
+            tty.warn("Package not installed, falling back to use stage dir")
+            tty.debug(f"stage dir: {package_metadata_root}")
+        else:
+            # Get the package's archived files
+            archive_files = []
+            archive_root = package_metadata_root / "archived-files"
+            if archive_root.is_dir():
+                archive_files = [str(f) for f in archive_root.rglob("*") if f.is_file()]
+            else:
+                msg = "Cannot copy package archived files: archived-files must be a directory"
+                tty.warn(msg)
+
     except spack.error.SpackError as e:
         tty.error(f"Cannot copy logs: {str(e)}")
         return
+    except AssertionError:
```

Comment:
what assertion?

Reply:
it is an assertion buried in `spec.package`

Reply:
Hm, I think AssertionError is too generic to catch. Can come from multiple places whereas here you assume a very specific place.

Reply:
The goal here is to a) suppress the exceptions we kind of expect and b) provide some reasonable error message scrubbing to the console.

Reply:
I suppose I could check if the spec is concrete before doing anything here. That is the main error I care about.

Reply:
exceptions all removed.